bundles/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/internal/handler/SystemInfoHandler.java

```diff
@@ -229,26 +226,22 @@ private boolean updateProperties() {
      * @return true if channel groups where added
      */
     private boolean addDynamicChannels() {
-        ThingUID thingUID = thing.getUID();
-
         List<ChannelGroupDefinition> newChannelGroups = new ArrayList<>();
-        newChannelGroups.addAll(createChannelGroups(thingUID, CHANNEL_GROUP_STORAGE, CHANNEL_GROUP_TYPE_STORAGE,
-                systeminfo.getFileOSStoreCount()));
-        newChannelGroups.addAll(createChannelGroups(thingUID, CHANNEL_GROUP_DRIVE, CHANNEL_GROUP_TYPE_DRIVE,
-                systeminfo.getDriveCount()));
-        newChannelGroups.addAll(createChannelGroups(thingUID, CHANNEL_GROUP_DISPLAY, CHANNEL_GROUP_TYPE_DISPLAY,
-                systeminfo.getDisplayCount()));
-        newChannelGroups.addAll(createChannelGroups(thingUID, CHANNEL_GROUP_BATTERY, CHANNEL_GROUP_TYPE_BATTERY,
-                systeminfo.getPowerSourceCount()));
-        newChannelGroups.addAll(createChannelGroups(thingUID, CHANNEL_GROUP_NETWORK, CHANNEL_GROUP_TYPE_NETWORK,
-                systeminfo.getNetworkIFCount()));
+        addChannelGroups(CHANNEL_GROUP_STORAGE, CHANNEL_GROUP_TYPE_STORAGE, systeminfo.getFileOSStoreCount(),
+                newChannelGroups);
+        addChannelGroups(CHANNEL_GROUP_DRIVE, CHANNEL_GROUP_TYPE_DRIVE, systeminfo.getDriveCount(), newChannelGroups);
+        addChannelGroups(CHANNEL_GROUP_DISPLAY, CHANNEL_GROUP_TYPE_DISPLAY, systeminfo.getDisplayCount(),
+                newChannelGroups);
+        addChannelGroups(CHANNEL_GROUP_BATTERY, CHANNEL_GROUP_TYPE_BATTERY, systeminfo.getPowerSourceCount(),
+                newChannelGroups);
+        addChannelGroups(CHANNEL_GROUP_NETWORK, CHANNEL_GROUP_TYPE_NETWORK, systeminfo.getNetworkIFCount(),
+                newChannelGroups);
         if (!newChannelGroups.isEmpty()) {
             logger.debug("Creating additional channel groups");
             newChannelGroups.addAll(0, thingTypeProvider.getChannelGroupDefinitions(thing.getThingTypeUID()));
-            ThingTypeUID thingTypeUID = new ThingTypeUID(BINDING_ID, THING_TYPE_COMPUTER_ID + idExtString);
-            if (thingTypeProvider.updateThingType(thingTypeUID, newChannelGroups)) {
+            if (thingTypeProvider.updateThingType(THING_TYPE_COMPUTER_IMPL, newChannelGroups)) {
```

Comment:
I don't think this condition is necessary. See my initial feedback. Removing this will allow you to remove the boolean return in `updateThingType`.

Reply:
Ok. I was able to remove more not used code

Reply:
Yeah. Done